### PR TITLE
feat(fsmv2): add AuthFailedState for permanent auth errors (ENG-4649)

### PR DESCRIPTION
**Pull request by JeremyTheocharis**
## Problem

When auth fails with InvalidToken or InstanceDeleted, the transport worker stays in `StartingState` forever — retrying at 60s (token) or 5min (deleted) backoff with no way out. The instance stops sending heartbeats (no JWT = no push children), so MC shows a stale "last seen" timestamp. Nothing in the logs indicates this needs a config change.

## Fix

New `AuthFailedState`. `StartingState` checks `LastErrorType` after a failure: if InvalidToken or InstanceDeleted, transition to `AuthFailedState` instead of entering backoff.

`AuthFailedState` is passive — dispatches no actions, just ticks. On each tick it compares three config fields between `snap.Desired` (current config) and `snap.Observed.FailedAuthConfig` (config at time of failure):

- `AuthToken`
- `RelayURL`
- `InstanceUUID`

When any field differs, it transitions back to `StartingState` for a fresh auth attempt. On shutdown, it transitions to `StoppingState`.

### Config detection via FailedAuthConfig

The failed auth config is stored in dependencies (`SetFailedAuthConfig`) and exposed via `CollectObservedState` as a `FailedAuthConfig` struct on `TransportObservedState`. This is necessary because the embedded `TransportDesiredState` in observed state is never populated with auth fields by the collector — comparing `snap.Observed.AuthToken` would always see empty strings.

`SetFailedAuthConfig` is called atomically with `RecordAuthError` inside `AuthenticateAction.Execute()` for permanent errors. `RecordSuccess()` clears it. `AuthFailedState` compares `snap.Desired.*` against `snap.Observed.FailedAuthConfig.*`.

### Stale error guard in StartingState

When `AuthFailedState` detects a config change and transitions to `StartingState`, the error state (`ConsecutiveErrors > 0`, `LastErrorType = InvalidToken`) persists in deps. Without a guard, `StartingState` would see the permanent error and transition right back to `AuthFailedState` without trying auth with the new config. The `authConfigChanged()` guard skips the permanent-error check when the desired config differs from the stored failed config.

### Architecture compliance

- Empty struct (embeds `helpers.StartingBase` only)
- Shutdown check first
- `PhaseStarting` lifecycle (children stay stopped — correct, no JWT)
- 10 new tests + added to all Architecture Compliance lists (64/64 pass)

## Stack

1. Suppress SentryError (#2483) → staging
2. Tiered alerting (#2485) → #2483
3. **→ This PR** → #2485

Part of ENG-4576. Resolves: ENG-4649.

## Test plan

- [x] `StartingState` → `AuthFailedState` on InvalidToken
- [x] `StartingState` → `AuthFailedState` on InstanceDeleted
- [x] `StartingState` does NOT transition on transient errors (Network, ServerError)
- [x] `AuthFailedState` → `StartingState` on AuthToken change
- [x] `AuthFailedState` → `StartingState` on RelayURL change
- [x] `AuthFailedState` → `StartingState` on InstanceUUID change
- [x] `AuthFailedState` stays when config unchanged
- [x] `AuthFailedState` → `StoppingState` on shutdown
- [x] Stale errors don't block retry after config change (P1 fix)
- [x] Architecture tests pass (64/64)
- [x] `go vet` clean
- [ ] Integration tests (CI)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by JeremyTheocharis** [comment]
@coderabbitai review

**Comment by JeremyTheocharis** [comment]
@claude review

File: docs/superpowers/specs/2026-03-18-fsmv2-workshop-nmap-worker-design.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+go run pkg/fsmv2/cmd/runner/main.go --scenario=connection --duration=30s
+```
+
+You should see: `Stopped → TryingToConnect → Up`. Kill nc — watch `Up → Down`. Restart nc — watch `Down → Up`.
+
+## Outlook
+
+- **Parent/children**: How the transport worker spawns push/pull children (`workers/transport/`)
+- **FSMv1 bridge**: Protocol Converter checks `ConnectionFSMState == "up"` — maps 1:1
+- **Why this matters**: ~800 lines of FSMv1 → ~200 lines of FSMv2, same behavior
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply docs/superpowers/specs/2026-03-18-fsmv2-workshop-nmap-worker-design.md]: Fixed — removed the workshop spec and all stale review artifacts in commit 52acbf83d. These were accidentally committed from a prior worktree session.

File: docs/superpowers/specs/2026-03-18-fsmv2-workshop-nmap-worker-design.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+**Lifecycle states** — handled by embedding a base type. You don't implement the lifecycle logic yourself:
+- `helpers.StoppedBase` — worker is off
+- `helpers.StartingBase` — worker is transitioning to active
+- `helpers.RunningHealthyBase` — worker is running and healthy
+- `helpers.RunningDegradedBase` — worker is running but impaired
+- `helpers.StoppingBase` — worker is gracefully shutting down
+
+**Your business logic** goes in `Next()`. Every state must:
+1. Check `IsShutdownRequested()` **first** — always, no exceptions
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply docs/superpowers/specs/2026-03-18-fsmv2-workshop-nmap-worker-design.md]: Fixed — file removed in 52acbf83d.

File: umh-core/003_klaus/artifacts/2026-03-11_eng4450_stack/pr3_06_silent_failure.md [hunk 3]
```diff
@@ -0,0 +1,445 @@
+# Silent Failure Audit: PR #2467 (Suppress Transient Transport Errors)
+
+**Auditor**: Silent Failure Hunter
+**PR**: feat(fsmv2): suppress transient transport errors from SentryError (4/4)
+**Branch**: `eng-4450-suppress-transient-errors`
+**Date**: 2026-03-11
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply umh-core/003_klaus/artifacts/2026-03-11_eng4450_stack/pr3_06_silent_failure.md]: Fixed — all stale review artifacts removed in 52acbf83d.

File: umh-core/pkg/fsmv2/workers/transport/state/state_test.go [hunk 4]
```diff
@@ -59,6 +59,11 @@ func makeSnapshotWithBackoff(shutdownRequested bool, desiredState string, jwtTok
 		LastErrorType:         lastErrorType,
 		LastAuthAttemptAt:     lastAuthAttemptAt,
 		LastRetryAfter:        lastRetryAfter,
+		FailedAuthConfig: snapshot.FailedAuthConfig{
+			AuthToken:    desired.AuthToken,
+			RelayURL:     desired.RelayURL,
+			InstanceUUID: desired.InstanceUUID,
+		},
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply umh-core/pkg/fsmv2/workers/transport/state/state_test.go:66]: Acknowledged. The `FailedAuthConfig` in `makeSnapshotWithBackoff` simulates the "config unchanged since failure" path. The dedicated `makeAuthFailedStartingSnapshot` helper covers the "config changed" path. Adding a clarifying comment to the helper would help — tracking as follow-up.

File: umh-core/pkg/fsmv2/workers/transport/state/starting.go [hunk 5]
```diff
@@ -72,6 +83,29 @@ func (s *StartingState) Next(snapAny any) fsmv2.NextResult[any, any] {
...
+
+// authConfigChanged returns true if the current desired auth config differs from the
+// config that was used in the last permanently-failed auth attempt. Used by StartingState
+// to skip stale permanent errors after a config change. AuthFailedState performs the same
+// comparison inline to capture per-field diagnostics in the reason string.
+func authConfigChanged(desired *snapshot.TransportDesiredState, observed snapshot.TransportObservedState) bool {
+	return desired.AuthToken != observed.FailedAuthConfig.AuthToken ||
+		desired.RelayURL != observed.FailedAuthConfig.RelayURL ||
+		desired.InstanceUUID != observed.FailedAuthConfig.InstanceUUID
+}
```
[the remark this answers is not in the document]
Comment (rpaland) [reply umh-core/pkg/fsmv2/workers/transport/state/starting.go:110]: Agree with this

**Review by rpaland — APPROVED** [review]
LGTM - That one claude comment should be addresses though

File: umh-core/pkg/fsmv2/workers/transport/state/starting.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply umh-core/pkg/fsmv2/workers/transport/state/starting.go:110]: Fixed in 3129b2a0e — added `IsEmpty()` guard at the top of `authConfigChanged()`:

```go
func authConfigChanged(desired *snapshot.TransportDesiredState, observed snapshot.TransportObservedState) bool {
    if observed.FailedAuthConfig.IsEmpty() {
        return false
    }
    return desired.AuthToken != observed.FailedAuthConfig.AuthToken ||
        desired.RelayURL != observed.FailedAuthConfig.RelayURL ||
        desired.InstanceUUID != observed.FailedAuthConfig.InstanceUUID
}
```

Empty `FailedAuthConfig` (the production state after transient errors) now returns `false` (no config change), so the backoff block applies correctly. Also added a regression test that builds a snapshot with intentionally empty `FailedAuthConfig` (not using `makeSnapshotWithBackoff` which masks the bug) and asserts backoff is respected.
